packages/form-config/src/components/Antd/UploadForm/index.tsx

```diff
@@ -0,0 +1,194 @@
+  const formId = match.params[ROUTE_PARAM_FORM_ID];
+  const formData = useSelector((state) => getManifestFilesById(state, formId));
+  let formInitialValues = initialValues;
+  if (formId && formData) {
+    formInitialValues = {
+      ...formInitialValues,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      form_name: formData.label,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      form_relation: formData.form_relation,
+      module: formData.module,
+    };
+  }
+  useEffect(() => {
+    if (formId) {
+      setIsEditMode(true);
+    }
+  }, [formId]);
+
+  const layout = {
+    labelCol: {
+      xs: { offset: 0, span: 16 },
+      sm: { offset: 2, span: 10 },
+      md: { offset: 0, span: 8 },
+      lg: { offset: 0, span: 6 },
+    },
+    wrapperCol: { xs: { span: 24 }, sm: { span: 14 }, md: { span: 12 }, lg: { span: 10 } },
+  };
+
+  const tailLayout = {
+    wrapperCol: {
+      xs: { offset: 0, span: 16 },
+      sm: { offset: 12, span: 24 },
+      md: { offset: 8, span: 16 },
+      lg: { offset: 6, span: 14 },
+    },
+  };
+  const { Title } = Typography;
+  const uploadProps = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onRemove: (file: any) => {
+      const index = fileList.indexOf(file);
+      const newFileList = fileList.slice();
+      newFileList.splice(index, 1);
+      setFileList(newFileList);
+    },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    beforeUpload: (file: any) => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      setFileList([...fileList, file]);
+      return false;
+    },
+  };
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const normFile = (e: any) => {
+    if (Array.isArray(e)) {
+      return e;
+    }
+    return e && e.fileList;
+  };
+
+  if (ifDoneHere && onSaveRedirectURL) {
+    return <Redirect to={onSaveRedirectURL} />;
+  }
+
+  return (
+    <div className="layout-content">
+      <Title level={3}>Upload Form</Title>
+      <Card>
+        <Form
+          {...layout}
+          initialValues={formInitialValues}
+          onFinish={(values) => {
+            submitForm(
+              {
+                ...values,
+                form: fileList,
+              },
+              accessToken,
+              opensrpBaseURL,
+              isJsonValidator,
+              setSubmitting,
+              setIfDoneHere
+            );
+          }}
+        >
+          <Form.Item
+            name="form_name"
+            label="Form Name"
+            rules={[{ required: true, message: 'Form Name is required' }]}
+          >
+            <Input disabled={isEditMode} />
+          </Form.Item>
+          <Form.Item name="module" label="Module">
+            <Input disabled={isEditMode} />
+          </Form.Item>
+          <Form.Item name="form_relation" label="Related to">
+            <Input disabled={isEditMode} />
```

Comment:
why don't this have any validation logic

Reply:
@p-netm They are not required. I am not also aware of any other validation rules that should be applied to these fields. The current version being used by reveal has no rules for these fields